File: trunk/C#/PokerWorld/Game/AbstractDealer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using EricUtility.Games.CardGame;

namespace PokerWorld.Game
{
    public abstract class AbstractDealer
    {
        protected Stack<GameCard> m_Deck;
       
        public AbstractDealer()
        {
            FreshDeck();
        }

        public abstract GameCard[] DealHoles(PlayerInfo p);
        public abstract GameCard[] DealFlop();
        public abstract GameCard DealTurn();
        public abstract GameCard DealRiver();

        public abstract void FreshDeck();
    }
}

File: trunk/JAVA/PokerWorld/src/bluffinmuffin/poker/entities/dealer/AbstractDealer.java
package bluffinmuffin.poker.entities.dealer;

import bluffinmuffin.game.entities.Card;
import bluffinmuffin.game.entities.CardSet;
import bluffinmuffin.poker.entities.PlayerInfo;

public abstract class AbstractDealer
{
    protected CardSet m_deck;
    
    public AbstractDealer()
    {
        freshDeck();
    }
    
    public abstract CardSet dealHoles(PlayerInfo p);
    
    public abstract CardSet dealFlop();
    
    public abstract Card dealTurn();
    
    public abstract Card dealRiver();
    
    public abstract void freshDeck();
}
